addons/udes_stock/models/product_product.py

```diff
@@ -14,19 +14,19 @@ class ProductProduct(models.Model):
 
     def assert_serial_numbers(self, serial_numbers):
         """
-        If the product in self is tracked by serial numbers, check if
+        If the product in self is tracked, check if
         the ones in serial_numbers are currently in use in the system.
         """
         Lot = self.env["stock.production.lot"]
         self.ensure_one()
-        if self.tracking == "serial":
+        if self.tracking != "none":
```

Comment:
Given that we're now asserting any kind of tracking, not just serial numbers, we should also change the name of the method e.g. to `assert_tracking_unique`. 
Q. In which case would we want a lot number to be unique?

Reply:
In all cases product_id lot number is unique for serial products and for products tracked with lot.
I agree about naming the method, you are right